File: bestia-models/src/test/java/net/bestia/model/geometry/PointTest.java
package net.bestia.model.geometry;

import java.io.Serializable;

import org.junit.Assert;
import org.junit.Test;

import net.bestia.model.geometry.Circle;
import net.bestia.model.geometry.Point;
import net.bestia.model.geometry.Rect;

public class PointTest {
	
	@Test
	public void is_serializable() {
		Assert.assertTrue(Serializable.class.isAssignableFrom(Point.class));
	}

	@Test
	public void moveAnchor() {
		Point p = new Point(10, 10);
		p = (Point) p.moveByAnchor(3, 3);
		Assert.assertEquals(3, p.getX());
		Assert.assertEquals(3, p.getY());
	}

	@Test
	public void collide_point_success() {
		Point p1 = new Point(10, 10);
		Point p2 = new Point(10, 10);
		
		Assert.assertTrue(p1.collide(p2));
	}

	@Test
	public void collide_point_fail() {
		Point p1 = new Point(10, 10);
		Point p2 = new Point(11, 10);
		
		Assert.assertFalse(p1.collide(p2));
	}

	@Test
	public void collide_circle_success() {
		Point p1 = new Point(10, 10);
		Circle c = new Circle(13, 10, 4);
		
		Assert.assertTrue(p1.collide(c));
	}

	@Test
	public void collide_circle_fail() {
		Point p1 = new Point(10, 10);
		Circle c = new Circle(13, 10, 3);
		
		Assert.assertTrue(p1.collide(c));
	}

	@Test
	public void collide_rect_success() {
		Point p1 = new Point(10, 10);
		Rect r = new Rect(9, 9, 5, 5);
		
		Assert.assertTrue(p1.collide(r));
	}

	@Test
	public void collide_rect_fail() {
		Point p1 = new Point(10, 10);
		Rect r = new Rect(11, 10, 5, 5);
		
		Assert.assertFalse(p1.collide(r));
	}

	@Test
	public void getAnchor() {
		Point p1 = new Point(10, 10);
		Assert.assertEquals(p1, p1.getAnchor());
	}
	
	@Test
	public void getBoundingBox() {
		Point p1 = new Point(10, 10);
		Assert.assertEquals(10, p1.getBoundingBox().getX());
		Assert.assertEquals(10, p1.getBoundingBox().getY());
		Assert.assertEquals(1, p1.getBoundingBox().getHeight());
		Assert.assertEquals(1, p1.getBoundingBox().getWidth());
	}
	
	
	public void getDistance() {
		// TODO
	}
	
	public void jackson_serialize() {
		// TODO
	}

}

File: bestia-models/src/test/kotlin/net/bestia/model/geometry/PointTest.kt
package net.bestia.model.geometry

import org.junit.jupiter.api.Assertions.*
import org.junit.jupiter.api.Test
import java.io.Serializable

class PointTest {

  @Test
  fun is_serializable() {
    assertTrue(Serializable::class.java.isAssignableFrom(Vec3::class.java))
  }

  @Test
  fun moveAnchor() {
    var p = Vec3(10, 10, 10)
    p = p.moveTo(3, 3, 3)
    assertEquals(3, p.x)
    assertEquals(3, p.y)
  }

  @Test
  fun collide_point_success() {
    val p1 = Vec3(10, 10, 10)
    val p2 = Vec3(10, 10, 10)

    assertTrue(p1.collide(p2))
  }

  @Test
  fun collide_point_fail() {
    val p1 = Vec3(10, 10, 10)
    val p2 = Vec3(11, 10, 10)

    assertFalse(p1.collide(p2))
  }

  @Test
  fun collide_circle_success() {
    val p1 = Vec3(10, 10, 10)
    val c = Sphere(13, 10, 10, 4)

    assertTrue(p1.collide(c))
  }

  @Test
  fun collide_circle_fail() {
    val p1 = Vec3(10, 10, 10)
    val c = Sphere(13, 10, 10, 3)

    assertTrue(p1.collide(c))
  }

  @Test
  fun collide_rect_success() {
    val p1 = Vec3(10, 10, 10)
    val r = Rect(9, 9, 9, 5, 5, 5)

    assertTrue(p1.collide(r))
  }

  @Test
  fun collide_rect_fail() {
    val p1 = Vec3(10, 10, 10)
    val r = Rect(11, 10, 10, 5, 5, 5)

    assertFalse(p1.collide(r))
  }

  @Test
  fun getAnchor() {
    val p1 = Vec3(10, 10, 10)
    assertEquals(p1, p1.anchor)
  }

  @Test
  fun getBoundingBox() {
    val p1 = Vec3(10, 10, 10)
    assertEquals(10, p1.boundingBox.x)
    assertEquals(10, p1.boundingBox.y)
    assertEquals(1, p1.boundingBox.height)
    assertEquals(1, p1.boundingBox.width)
  }
}
